test/suites/drift/suite_test.go

```diff
@@ -335,5 +337,67 @@ var _ = Describe("Drift", Label("AWS"), func() {
 			}}),
 			Entry("Start-up Taints Drift", "Start-up Taint", test.ProvisionerOptions{StartupTaints: []v1.Taint{{Key: "example.com/another-taint-2", Effect: v1.TaintEffectPreferNoSchedule}}}),
 		)
+		DescribeTable("AWSNodeTemplate Drift", func(fieldName string, awsnodetemplatespec v1alpha1.AWSNodeTemplateSpec) {
+			if fieldName == "InstanceProfile" {
+				By("creating an instance profile")
+				awsnodetemplatespec.AWS.InstanceProfile = awssdk.String(*awsnodetemplatespec.AWS.InstanceProfile + "-" + settings.FromContext(env.Context).ClusterName)
```

Comment:
Is this the right name? It looks like we are just appending the cluster name to the current instance profile name, which I think is different from what you have in the IAM policy

Reply:
The instance profile name will be `instance-profile-drift-${CLUSTER NAME}`